csharp/source/autowrap/csharp/csharp.d

```diff
@@ -136,24 +139,22 @@ public struct csharpRange {
 }
 
 public string generateCSharp(Modules...)(LibraryName libraryName, RootNamespace rootNamespace) if(allSatisfy!(isModule, Modules)) {
+    import core.time : Duration;
+    import std.datetime : Date, DateTime, SysTime, TimeOfDay, TimeZone;
     import autowrap.reflection : AllAggregates;
     import std.traits : moduleName;
-    generateSliceBoilerplate(libraryName.value);
-
-    foreach(agg; AllAggregates!Modules) {
-        alias modName = moduleName!agg;
-        const string aggName = __traits(identifier, agg);
-        CSharpAggregate csagg = getAggregate(getCSharpName(modName), getCSharpName(aggName), !is(agg == class));
-
-        generateRangeDef!agg(libraryName.value);
-
-        generateConstructors!agg(libraryName.value, csagg);
 
-        generateMethods!agg(libraryName.value, csagg);
-
-        generateProperties!agg(libraryName.value, csagg);
+    generateSliceBoilerplate(libraryName.value);
 
-        generateFields!agg(libraryName.value, csagg);
+    alias aggregates = AllAggregates!Modules;
+    static foreach(agg; aggregates) {
+        static if (!(is(agg == Date) || is(agg == DateTime) || is(agg == SysTime) || is(agg == TimeOfDay) || is(agg == Duration) || is(agg == TimeZone))) {
+            generateRangeDef!agg(libraryName.value);
+            generateConstructors!agg(libraryName.value, getAggregate(getCSharpName(moduleName!agg), getCSharpName(__traits(identifier, agg)), !is(agg == class)));
+            generateMethods!agg(libraryName.value, getAggregate(getCSharpName(moduleName!agg), getCSharpName(__traits(identifier, agg)), !is(agg == class)));
+            generateProperties!agg(libraryName.value, getAggregate(getCSharpName(moduleName!agg), getCSharpName(__traits(identifier, agg)), !is(agg == class)));
+            generateFields!agg(libraryName.value, getAggregate(getCSharpName(moduleName!agg), getCSharpName(__traits(identifier, agg)), !is(agg == class)));
```

Comment:
These 4 lines are long and differ only very slightly.

Reply:
Yup, because of CTFE rules the result of getAggregate as stored in a variable cannot be read at compile time.

Reply:
And storing it in an `alias` or `enum` also doesn't work for some reason?

Reply:
The error is that __traits(identifier) is not valid in alias.

Reply:
It is in the newest dmd, and before that one could always use `std.meta.Alias` to get around that.

Reply:
I guess I'll have to upgrade my local DMD then. Tomorrow.

Reply:
Just tried `alias aggid = __traits(identifier, agg);` with 2.084.0 and got this error: `Error: trait identifier is either invalid or not supported in alias`
This does work: `const string aggid = __traits(identifier, agg);`

Reply:
Huh, weird. `alias aggid = Alias!(__traits(identifier, agg);` has always worked though.

Reply:
I came up with a better way of handling this. See latest commit.